src/client.rs

```diff
@@ -271,10 +273,31 @@ impl Response {
         &self.inner.headers
     }
 
+    // This function is needed to overcome borrow checker rules when returning a JSON response.
+    fn headers_clone(&self) -> Headers {
+        self.inner.headers.clone()
+    }
+
     /// Get the `HttpVersion`.
     pub fn version(&self) -> &HttpVersion {
         &self.inner.version
     }
+
+    /// Try and deserialize the response body as JSON.
+    pub fn json<T: Deserialize>(&mut self) -> ::Result<T> {
+        let cloned_headers = self.headers_clone();
```

Comment:
Instead of cloning the headers, which can mean cloning quite a few strings, we could just use the mutable borrow after the match, instead of inside it.

```rust
match self.inner.headers.get::<ContentType>() {
  Some(ty) => {},
  None => return Err(::Error::Serialize("Content-Type is not application/json".into()))
}

serde_json::from_reader(...)
```